AgrirouterClientTester/src/Application.cpp

```diff
@@ -398,12 +422,31 @@ void Application::onMessageCallback(int event, Response *response, std::string a
             printf("ListEndpointsResponse: %s\n", msg.c_str());
         }
     }
+
+    self->m_running = false;
 }
 
-void Application::onErrorCallback(int statusCode, int connectionProviderErrorCode, std::string curlMessage,
-                                    std::string applicationMessageId, std::string content, void *callbackCallee)
+void Application::onErrorCallback(int statusCode, int connectionProviderErrorCode, std::string errorMessage,
+                                    std::string applicationMessageId, std::string errorContent, void *callbackCallee)
 {
+    Application *self = static_cast<Application *>(callbackCallee);
     printf("\n");
-    printf("On Error callback: http code: %i, CURL code: %i, CURL message: %s, Application message id: %s, error page content: %s\n",
-                statusCode, connectionProviderErrorCode, curlMessage.c_str(), applicationMessageId.c_str(), content.c_str());
+    printf("On Error callback: http code: %i, CURL code: %i, CURL message: %s, Application message id: %s, error page errorContent: %s\n",
+                statusCode, connectionProviderErrorCode, errorMessage.c_str(), applicationMessageId.c_str(), errorContent.c_str());
+    
+    self->m_running = false;
+}
+
+void Application::onLogCallback(int logLevel, std::string logMessage, void *callbackCallee)
+{
+    if(logLevel < MG_LFL_TRC)
```

Comment:
Maybe it would be better to have the possibility to configure the minimum log level.